### DISCUSSION TITLE
Error when eager loading many to many relation with BelongsToManyMultiSelect field

### DISCUSSION THREAD

**Discussion by mohamedmoussasoli** [filamentphp/filament#2313] [Help]
### Package

filament/filament

### Package Version

v2.10.41

### Laravel Version

v9.2

### PHP Version

8.1.2

### Bug description

I have MorphToMany relation

`/**
     * @return \Illuminate\Database\Eloquent\Relations\MorphToMany|CategoryBuilder
     */
    public function categories ()
    {

        return $this->morphToMany(
            Category::class,
            'model',
            'model_has_categories'
        );

    }`

and BelongsToManyMultiSelect field 

`BelongsToManyMultiSelect::make('categories')
        ->label(__('admin.Categories'))
        ->required()
        ->preload()
        ->relationship('categories', 'name'),`

when i eager load the categories relation using the $with property 
`protected $with = ['categories'];`

i get this error
<img width="1279" alt="Screenshot 2022-04-19 at 1 47 49 PM" src="https://user-images.githubusercontent.com/9276573/163996719-e1ff72b2-f0d2-4ed5-8f0b-7d0f508f0675.png">

**Comment by danharrin** [comment]
Sorry, I cannot reproduce this. I assume this is probably an issue with your database setup rather than Filament?
